src/itp.rst

```diff
@@ -0,0 +1,218 @@
+.. :Authors: - Ke Du
+
+.. title:: Interactive Theorem Provers
+
+Overview
+========
+Formal verification is essential to make sure important properties such as safety hold in software or hardware systems. 
+  
+
+
+  :cite:`klein09` Complete formal verification is the only known way to guarantee that a system is free of programming errors. 
+
+Important applications of formal verification include: aerospace softwares such as those who prevent airplane collision;
+microprocessor producers such as Intel have their own verification team; there are also works about the
+language Rust, which, though haven't been completely verified, provides safety and performance at the same time. 
+A cryptographic system must ensure that its protocol is error free, while the errors are hard to notice.
+
+An interactive theorem prover (ITP), sometimes refered to as proof assistant, is a programming environment that helps 
+humans to write formal proofs. In contrast to automated theorem provers, ITPs require human guidance, 
+though they also have a certain level of automation power. 
+Different to a traditional proof conducted through a pen-and-paper approach, a formal proof conducted by computers
+is completely rigorous, and is oftenly tedious and hard.
+
+It is worth noting that, when we say some formal proof done by a theorem prover works, 
+we impose the implicit assumption that the theorem prover is error-free. In other words, we have to trust the theorem provers.
```

Comment:
use the phrase "trusted computing base (TCB)" here, since that shows up in a lot of papers